core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -174,7 +171,8 @@ public void start(Listener<RespT> observer, Metadata headers) {
     }
 
     if (stream == null) {
-      stream = new DelayedStream(transportFuture, headers, listener);
+      stream = new DelayedStream(
+          this, callOptions, transportFuture, headers, listener, callExecutor, method);
```

Comment:
`this` is a different lock than before. Why not just create a lock in DelayedStream or use the intrinsic lock like before?